src/polymer/iron-flex-layout-classes.ts

```diff
@@ -0,0 +1,228 @@
+          range: sourceRangeForAddAttribute(parsedDocument, styleNode)
+        }];
+      }
+      warnings.push(warning);
+    }
+    const body = dom5.query(parsedDocument.ast, p.hasTagName('body'));
+    // Handle files like `<dom-module></dom-module> <body><p>hello</p></body>`
+    // where a "fake" body node would be created by dom-module. Skip these
+    // cases, dear user please write proper HTML ¯\_(ツ)_/¯
+    if (!body || !body.__location) {
+      return warnings;
+    }
+    const missingModules = getMissingStyleModules(body);
+    if (!missingModules) {
+      return warnings;
+    }
+    // TODO(valdrin): update the warning location to be at the spot where
+    // the class is used.
+    const warning = createWarning(parsedDocument, body, missingModules);
+    const indent = getIndentationInside(body);
+    warning.fix = [{
+      replacementText: `
+${indent}<custom-style>
+${indent}  <style is="custom-style" include="${missingModules}"></style>
+${indent}</custom-style>`,
+      range: sourceRangeForPrependContent(parsedDocument, body)
+    }];
+    warnings.push(warning);
+    return warnings;
+  }
+}
+
+function getMissingStyleModules(node: dom5.Node) {
+  let includes = '';
+  for (const style of dom5.queryAll(node, isStyleInclude)) {
+    includes += ' ' + dom5.getAttribute(style, 'include')!;
+  }
+  return styleModules
+      .map((m) => {
+        if (includes.indexOf(m.name) === -1 && dom5.query(node, m.selector)) {
+          return m.name;
+        }
+      })
+      .filter((m) => !!m)
+      .join(' ');
+}
+
+function createWarning(
+    parsedDocument: ParsedHtmlDocument,
+    node: dom5.Node,
+    missingModules: string) {
+  const multi = missingModules.indexOf(' ') > -1;
+  return new FixableWarning({
+    code: 'iron-flex-layout-classes',
+    message: `Style module${multi ? 's are' : ' is'} used but not imported:
+
+  ${missingModules}
+
+Import ${multi ? 'them' : 'it'} in the template style include.`,
+    parsedDocument,
+    severity: Severity.WARNING,
+    sourceRange: parsedDocument.sourceRangeForStartTag(node)!
+  });
+}
+
+function getStyleNodeToEdit(node: dom5.Node) {
+  let styleToEdit = null;
+  for (const style of dom5.queryAll(node, isStyleInclude)) {
+    // Get the first one of the styles with include attribute, otherwise
+    // prefer styles that already include iron-flex-layout modules.
+    if (!styleToEdit ||
+        styleModulesRegex.test(dom5.getAttribute(style, 'include')!)) {
+      styleToEdit = style;
+    }
+  }
+  // Fallback to style without include attribute.
+  return styleToEdit || dom5.query(node, p.hasTagName('style'));
+}
+
+function sourceRangeForAddAttribute(
+    parsedDocument: ParsedHtmlDocument, node: dom5.Node) {
+  const tagRange = parsedDocument.sourceRangeForStartTag(node)!;
+  return {
+    file: tagRange.file,
+    start: {line: tagRange.end.line, column: tagRange.end.column - 1},
+    end: {line: tagRange.end.line, column: tagRange.end.column - 1}
```

Comment:
This -1 also makes me nervous for unquoted attributes. Mind adding a test for that?

Reply:
The range is for the start tag, so the -1 will go right before the closing >, e.g. in `<div>` it would go to `<div`